stan/math/opencl/matrix_cl.hpp

```diff
@@ -56,13 +114,16 @@ class matrix_cl {
       return;
     // the context is needed to create the buffer object
     cl::Context& ctx = opencl_context.context();
+    cl::CommandQueue queue = opencl_context.queue();
     try {
       // creates a read&write object for "size" double values
       // in the provided context
       oclBuffer_ = cl::Buffer(ctx, CL_MEM_READ_WRITE, sizeof(double) * size());
-
-      opencl_kernels::copy(cl::NDRange(rows_, cols_), A.buffer(),
-                           this->buffer(), rows_, cols_);
+      cl::Event cstr_event;
+      queue.enqueueCopyBuffer(A.buffer(), this->buffer(), 0, 0,
+                              A.size() * sizeof(double), &A.read_write_events(),
```

Comment:
I'm realizing there are a lot of places where you take the reference of the event vectors but I think they're always already references via those getters you made, right?

Reply:
So I tried removing that `&` but the compiler throws on that. I'm not sure if we are guranteed to get a ref from those functions